client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java

```diff
@@ -126,13 +125,7 @@ public class ShuffleClientImpl extends ShuffleClient {
   private final boolean authEnabled;
   private final TransportConf dataTransportConf;
 
-  private final ThreadLocal<Compressor> compressorThreadLocal =
-      new ThreadLocal<Compressor>() {
-        @Override
-        protected Compressor initialValue() {
-          return Compressor.getCompressor(conf);
-        }
-      };
+  private ThreadLocal<Compressor> compressorThreadLocal;
 
```

Comment:
The change in this class does not look related to error-prone.
Do we want to split it out ?

Reply:
@mridulm, this change follows [ThreadLocalUsage](https://errorprone.info/bugpattern/ThreadLocalUsage).